### ISSUE TITLE
Grave keys always pre-enchanted

### ISSUE THREAD

**Issue by Alphanos** [Corail31/tombstone_lite#29]
Minecraft: 1.12.2
Corail Tombstone: 3.2.5-1.12
Forge: 2772

In testing a variety of config options, I'm finding that no matter what I do, every grave key I receive upon death comes pre-enchanted to allow teleporting to the grave.  Here are the options I have tried using to prevent this:

    I:chanceEnchantedGraveKey=0

This comes with a chance of zero by default.  I tried raising it to a small non-zero number in case it was bugged when set to zero.

    B:tombstone.item.grave_key.name=true

I tried setting this to false in an attempt to completely disable gravestone keys from being enchanted at all, but this didn't help.

    B:enableHalloweenEffect=true
    B:enableHalloweenGrave=true

I know you mentioned that the new knowledge/perk system had some bonuses activated for Halloween, so I tried disabling the Halloween options in case they were causing the keys to all be enchanted.

=======

As a separate and less important matter, I also tried to disable the Corail Tombstone's Soulbound enchantment to reduce confusion from overlap with enchants from other mods:

    B:enableEnchantmentSoulbound=false

This did not seem to prevent the enchantment from being visible in JEI, but I'm not sure if that may be normal - maybe the config option is only intended to remove the enchant from the table that random enchantment selection draws from.  Mainly I mention this in case there is some connection with the enchanted grave keys - e.g. trouble reading larger portions of the config file.

Thanks!

**Comment by Corail31** [comment]
Yes, it's normal : during Halloween, as it's a period of magic too (the Sabbat), the grave's keys are always enchanted on respawn, and you also receive ghostly shape 5 (if the duration in the config is above 0 indeed).

**Comment by Alphanos** [comment]
Thanks very much for the fast and detailed reply =).

I had been trying to disable the pre-enchanted grave keys for balance purposes, but if this is a temporary event, then that's okay =).

I suppose that to avoid people running into confusion about such seasonal events, one possible suggestion would be to change the current "always" enchanted setup of the event to a multiplier.

E.g. in the config file, it could define the current:

     I:chanceEnchantedGraveKey=10

Which could then be modified by:

     I:chanceEnchantedGraveKeySeasonalEventMultiplier=5

So during any seasonal events like Halloween, the chance of a key being pre-enchanted would rise from the regular 10% to a temporary 50%.  I'm very happy with the mod regardless, but since you asked for feedback I just thought it could be worth suggesting.

I also just wanted to mention, when I tested various different gravestone mods out there, I found yours to be the best by far.  Fantastic cross-mod support for other mods that mess around with the inventory or custom slots, support for grave recovery by either shifting or right-clicking, items going back into their slots when possible, items going directly into the inventory when possible instead of potentially falling on the ground into lava/void, the key highlighting visually where to find the grave if it's lost in confusing tunnels, and the fact that you can optionally give the key to another player if they're helping you to recover your grave contents.  Then there is icing on the cake with great grave visuals, multiple decorative options to choose from, and a whole grave/death-themed magic mod built in.

Thanks very much for your work on such a great mod =)!

**Comment by Corail31** [comment]
For ballancing purposes, so you can apply the config option about nerfGhostlyShape : it'll be applied to many things (when opening a chest, harvesting a block, colliding a hostile creature, the teleportation will reduce the duration to 10 sec)
Events won't have the same bonus & content, so i was more in the idea to only display in the gui "knowledge of death" the bonus when an event is active
NB : i had to publish the content one month sooner to be sure it would be in modpack in time ;)
